Dash/Dash_Deploy_app_in_Naas.ipynb

```diff
@@ -0,0 +1,385 @@
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import dash\n",
+    "from dash import html\n",
+    "from dash import dcc\n",
+    "import plotly.graph_objects as go\n",
+    "import plotly.express as px\n",
+    "import os"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "aggressive-trustee",
+   "metadata": {},
+   "source": [
+    "### Variables"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "continuous-melbourne",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2022-07-31T19:05:11.979770Z",
+     "iopub.status.busy": "2022-07-31T19:05:11.979544Z",
+     "iopub.status.idle": "2022-07-31T19:05:11.982616Z",
+     "shell.execute_reply": "2022-07-31T19:05:11.981993Z",
+     "shell.execute_reply.started": "2022-07-31T19:05:11.979748Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "DASH_PORT = 8050"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "registered-showcase",
+   "metadata": {},
+   "source": [
+    "## Model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "tested-astrology",
+   "metadata": {},
+   "source": [
+    "### Create app layout"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3cb8edf6-bf0e-46ad-bbea-f0206b531987",
+   "metadata": {},
+   "source": [
+    "### Initialize Dash app"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "crude-louisville",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2022-07-31T19:05:29.585741Z",
+     "iopub.status.busy": "2022-07-31T19:05:29.585508Z",
+     "iopub.status.idle": "2022-07-31T19:05:29.594514Z",
+     "shell.execute_reply": "2022-07-31T19:05:29.593793Z",
+     "shell.execute_reply.started": "2022-07-31T19:05:29.585719Z"
+    },
+    "papermill": {},
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "app = dash.Dash(requests_pathname_prefix=f'/user/{os.environ.get(\"JUPYTERHUB_USER\")}/proxy/{DASH_PORT}/')   \n",
+    "#app = dash.Dash() if you are not in Naas"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9baacf1d-7dbd-4398-8f63-9de552af0043",
+   "metadata": {},
+   "source": [
+    "### Get stock prices and create chart"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "c7bb0592-3522-49e7-be1f-e4706dd18b8a",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2022-07-31T19:05:32.549036Z",
+     "iopub.status.busy": "2022-07-31T19:05:32.548761Z",
+     "iopub.status.idle": "2022-07-31T19:05:32.589042Z",
+     "shell.execute_reply": "2022-07-31T19:05:32.588289Z",
+     "shell.execute_reply.started": "2022-07-31T19:05:32.549012Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "df = px.data.stocks() #reading stock price dataset \n",
+    "def stock_prices():\n",
+    "    # Function for creating line chart showing Google stock prices over time \n",
+    "    fig = go.Figure([go.Scatter(x = df['date'], y = df['GOOG'],\\\n",
```

Comment:
It's not clear where `df` is defined, and where it gets its values from? 
Maybe there should be a line before that defining it? Unless I'm missing something?

Reply:
defined in the previous line ;)